bundles/org.openhab.binding.modbus.e3dc/src/main/java/org/openhab/binding/modbus/e3dc/internal/handler/E3DCWallboxThingHandler.java

```diff
@@ -0,0 +1,233 @@
+            } else {
+                logger.warn("Bridge null");
+            }
+        });
+    }
+
+    @Override
+    public void dispose() {
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (command instanceof OnOffType) {
+            int writeValue = 0;
+            synchronized (this) {
+                if (channelUID.getIdWithoutGroup().equals(WB_SUNMODE_CHANNEL)) {
+                    currentBitSet.set(WB_SUNMODE_BIT, command.equals(OnOffType.ON));
+                } else if (channelUID.getIdWithoutGroup().equals(WB_CHARGING_ABORTED_CHANNEL)) {
+                    currentBitSet.set(WB_CHARGING_ABORTED_BIT, command.equals(OnOffType.ON));
+                } else if (channelUID.getIdWithoutGroup().equals(WB_SCHUKO_ON_CHANNEL)) {
+                    currentBitSet.set(WB_SCHUKO_ON_BIT, command.equals(OnOffType.ON));
+                } else if (channelUID.getIdWithoutGroup().equals(WB_1PHASE_CHANNEL)) {
+                    currentBitSet.set(WB_1PHASE_BIT, command.equals(OnOffType.ON));
+                }
+                writeValue = DataConverter.toInt(currentBitSet);
+                logger.debug("Wallbox write {}", writeValue);
+            }
+            int wallboxId = getWallboxId(config);
+            if (wallboxId != -1) {
+                wallboxSet(wallboxId, writeValue);
+            }
+        }
+    }
+
+    /**
+     * Wallbox Settings can be changed with one Integer
+     *
+     * @param wallboxId needed to calculate right register
+     * @param writeValue integer to be written
+     */
+    public void wallboxSet(int wallboxId, int writeValue) {
+        E3DCThingHandler localBridgeHandler = bridgeHandler;
+        if (localBridgeHandler != null) {
+            ModbusCommunicationInterface comms = localBridgeHandler.getComms();
+            if (comms != null) {
+                ModbusRegisterArray regArray = new ModbusRegisterArray(writeValue);
+                ModbusWriteRegisterRequestBlueprint writeBluePrint = new ModbusWriteRegisterRequestBlueprint(
+                        localBridgeHandler.getSlaveId(), WALLBOX_REG_START + wallboxId, regArray, false, 3);
+                comms.submitOneTimeWrite(writeBluePrint, result -> {
+                    if (dataWrite != ReadWriteSuccess.SUCCESS) {
+                        dataWrite = ReadWriteSuccess.SUCCESS;
+                        updateStatus();
+                    }
+                    logger.debug("E3DC Modbus write response! {}", result.getResponse().toString());
+                }, failure -> {
+                    if (dataWrite != ReadWriteSuccess.FAILED) {
+                        dataWrite = ReadWriteSuccess.FAILED;
+                        updateStatus();
+                    }
+                    logger.warn("E3DC Modbus write error! {}", failure.getRequest().toString());
+                });
+            }
+        }
+    }
+
+    private int getWallboxId(@Nullable E3DCWallboxConfiguration c) {
```

Comment:
Small suggestion : utilize OptionalInt instead of the "magic value" -1

Reply:
Ok, changed return value and adapted function references